catalog/src/main/java/org/killbill/billing/catalog/DefaultCatalogService.java

```diff
@@ -102,6 +136,36 @@ public StaticCatalog getCurrentCatalog(final InternalTenantContext context) thro
     }
 
     private VersionedCatalog getCatalog(final InternalTenantContext context) throws CatalogApiException {
-        return catalogCache.getCatalog(context);
+        if (pluginCatalog == null) {
+            return catalogCache.getCatalog(context);
```

Comment:
There could potentially be multiple catalog plugins, and maybe not all of them know about that specific tenant. You could go through all plugins and return the first non-null catalog, defaulting to the XML if none is found.